feat: add socket to OperatorInfo struct (#463)

pablodeymo opened the pull request:
### What Changed?
This PR adds a missing field `socket` to `OperatorInfo`. We also added a method `get_operator_socket()` in `AvsRegistryServiceChainCaller`.

The new method is used in `get_operators_avs_state_at_block()`, now you can obtain the operator socket value that is inside the `OperatorInfo`.

### Reviewer Checklist

- [ ] New features are tested and documented
- [ ] PR updates the changelog with a description of changes
- [ ] PR has one of the `changelog-X` labels (if applies)
- [ ] Code deprecates any old functionality before removing it

Fixes #

### What Changed?
<!-- Describe the changes made in this pull request -->

### Reviewer Checklist

- [ ] New features are tested and documented
- [ ] PR updates the changelog with a description of changes
- [ ] PR has one of the `changelog-X` labels (if applies)
- [ ] Code deprecates any old functionality before removing it

damiramirez committed:
feat: add socket to OperatorInfo struct (#463)

### What Changed?
This PR adds a missing field `socket` to `OperatorInfo`. We also added a
method `get_operator_socket()` in `AvsRegistryServiceChainCaller`.

The new method is used in `get_operators_avs_state_at_block()`, now you
can obtain the operator socket value that is inside the `OperatorInfo`.

### Reviewer Checklist

- [ ] New features are tested and documented
- [ ] PR updates the changelog with a description of changes
- [ ] PR has one of the `changelog-X` labels (if applies)
- [ ] Code deprecates any old functionality before removing it

damiramirez commented:
Docs code don't compile. Fixed [here](https://github.com/Layr-Labs/eigensdk-rs/pull/466)

damiramirez committed:
fix: make docs compile (#466)

### What Changed?
Compile `AvsRegistryService` docs

### Reviewer Checklist

- [ ] New features are tested and documented
- [ ] PR updates the changelog with a description of changes
- [ ] PR has one of the `changelog-X` labels (if applies)
- [ ] Code deprecates any old functionality before removing it

MegaRedHand commented on CHANGELOG.md:
````diff
@@ -100,6 +100,48 @@ Alternate implementation which directly queries from middleware using view call
         .unwrap();
   ```
 
+* Added field `socket` to `OperatorInfo` and a method `get_operator_socket` to retrieve the socket from the `AvsRegistryServiceChainCaller` in PR [463](https://github.com/Layr-Labs/eigensdk-rs/pull/463).
````
We should split this into two: "Added field `socket` to `OperatorInfo`" is a breaking change, while "Added a method `get_operator_socket` to retrieve the socket" goes in the "Added" section.

damiramirez committed:
refactor: update CHANGELOG for #464 (#468)

### What Changed?
This PR fixes a comment made in #464 and adds documentation for the
`get_operator_socket` method.

### Reviewer Checklist

- [ ] New features are tested and documented
- [ ] PR updates the changelog with a description of changes
- [ ] PR has one of the `changelog-X` labels (if applies)
- [ ] Code deprecates any old functionality before removing it

---------

Co-authored-by: Tomás Grüner <47506558+MegaRedHand@users.noreply.github.com>

MegaRedHand commented:
@dralves Could you approve this again, please? We made some changes to the docs and changelog, and your approval was dismissed.
